qt/scientific_interfaces/Inelastic/Analysis/IndirectDataAnalysisTab.h

```diff
@@ -35,13 +37,36 @@ class MANTIDQT_INELASTIC_DLL IndirectDataAnalysisTab : public IndirectTab {
   Q_OBJECT
 
 public:
-  IndirectDataAnalysisTab(IndirectFittingModel *model, QWidget *parent = nullptr);
+  IndirectDataAnalysisTab(std::string const &tabName, bool const hasResolution, QWidget *parent = nullptr);
   virtual ~IndirectDataAnalysisTab() override = default;
 
-  void setFitDataPresenter(std::unique_ptr<IndirectFitDataPresenter> presenter);
-  void setPlotView(IIndirectFitPlotView *view);
-  void setOutputOptionsView(IIndirectFitOutputOptionsView *view);
-  void setFitPropertyBrowser(IndirectFitPropertyBrowser *browser);
+  template <typename FittingModel> void setupFittingModel() { m_fittingModel = std::make_unique<FittingModel>(); }
```

Comment:
I think it is better to move these functions to the cpp file to keep the header file minimal

Reply:
I have moved the non-templated setup function to the cpp file. The templated setup member functions cannot be implemented in the cpp file because the compiler requires to see the entire definition of a templated function, and it can also see if when it is in the header file. I am hoping I can tidy this up, or find a solution, as part of #36390

Reply:
Ok, thanks for the explanation!